.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,23 @@
+name: Java CI with Maven for checkstyle plugin
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build with Maven
+      run: mvn clean install checkstyle:checkstyle
```

Comment:
@vpetrovaa why do we need to install it?
we can use checkstyle within it's maven plugin